commit c5f212171fce0b079b7373865318154caeb67c8b (refs/changes/81/3281/1)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-01-31T11:23:38-08:00 (1 year, 8 months ago)
    
    TRITON-96 cn-agent heartbeater crash loading VMs: "vmobjs ([object]) is required"

diff --git a/lib/heartbeater.js b/lib/heartbeater.js
index fc24431..a9a8da6 100644
--- a/lib/heartbeater.js
+++ b/lib/heartbeater.js
@@ -118,14 +118,14 @@ StatusReporter.prototype.updateSample = function () {
                 }, function onLoadVms(err, vmobjs) {
                     var vmobj;
 
-                    assert.arrayOfObject(vmobjs, 'vmobjs');
-
                     if (err) {
                         self.log.error({err: err}, 'unable update VM list');
                         cb(new Error('unable to update VM list.'));
                         return;
                     }
 
+                    assert.arrayOfObject(vmobjs, 'vmobjs');
+
                     vms = {};
                     newSample.vms = {};
 
